src/form/Form.tsx

```diff
@@ -1053,29 +1056,88 @@ function Form({
     }
   };
 
-  function handleActions(setLoader: any) {
-    for (let i = 0; i < activeStep.servar_fields.length; i++) {
-      const servar = activeStep.servar_fields[i].servar;
-      const fieldVal = fieldValues[servar.key];
-      if (servar.type === 'login') {
-        setLoader();
-        // Unless we want to do something more complex here we need to make a
-        // choice and can't just return both. Prioritize stytch
-        // @ts-expect-error
-        if (integrations.stytch) {
-          return sendMagicLink(fieldVal);
-        } else {
-          return sendLoginCode(fieldVal, servar);
+  // usePayments /Stripe
+  const [setCardElement, setupPaymentMethod] = usePayments();
+
+  // Action config that groups actions by servar type, orders them, configures behavior, etc.
+  interface ActionData {
+    fieldVal: any;
+    servar: any;
+    client: any;
+    formattedFields: any;
+    updateFieldValues: any;
+    step: any;
+    integrationData: any;
+  }
+  const actionConfigurations = useMemo(
+    () => [
+      {
+        servarType: 'payment_method',
+        integrationKey: 'stripe',
+        actionFn: setupPaymentMethod,
+        continue: true
+      },
+      {
+        servarType: 'login',
+        integrationKey: 'stytch',
+        actionFn: ({ fieldVal }: ActionData) => sendMagicLink(fieldVal),
+        continue: false
+      },
+      {
+        servarType: 'login',
+        integrationKey: 'firebase',
+        actionFn: ({ fieldVal, servar }: ActionData) =>
+          sendLoginCode(fieldVal, servar),
+        continue: false
+      },
+      {
+        servarType: 'pin_input',
+        integrationKey: 'firebase',
+        isMatch: ({ servar }: ActionData) => servar.metadata.verify_sms_code,
+        actionFn: ({ fieldVal, servar, client }: ActionData) =>
```

Comment:
you can just modify the helper functions directly to take in ActionData instead of having an intermediate actionFn parse it

Reply:
These functions are called from elsewhere as well.  I was trying to avoid the regression test effort by not changing the signatures, but I guess I will just do it.